ui-components/src/helpers/filters.ts

```diff
@@ -39,21 +42,42 @@ export function encodeToFrontendFilterString(filterParams: ListingFilterParams)
   let queryString = ""
   for (const filterType in filterParams) {
     const value = filterParams[filterType]
-
     if (filterType in ListingFilterKeys && value !== undefined && value !== "") {
       queryString += `&${filterType}=${value}`
     }
   }
   return queryString
 }
 
+export enum AvailabilityFilterType {
+  any = "any",
+  hasAvailability = "hasAvailability",
+  noAvailability = "noAvailability",
+  waitlist = "waitlist",
+}
+
+export type FormFilterData = ListingFilterParams & {
+  availability?: AvailabilityFilterType
+}
+
 export function decodeFiltersFromFrontendUrl(query: ParsedUrlQuery) {
-  const filters: ListingFilterParams = {}
+  const filters: FormFilterData = {}
   let foundFilterKey = false
   for (const queryKey in query) {
-    if (queryKey in ListingFilterKeys) {
-      filters[queryKey] = query[queryKey]
-      foundFilterKey = true
+    if (!(queryKey in ListingFilterKeys)) continue
```

Comment:
You've bundled some other small refactors here that would be nice to see separately next time.

Reply:
Ack